middleware/cache/heap.go

```diff
@@ -0,0 +1,92 @@
+package cache
+
+import (
+	"container/heap"
+)
+
+type heapEntry struct {
+	key   string
+	exp   uint64
+	bytes uint
```

Comment:
Entry size is stored in the heap to update the total size without reading entries on delete